src/FlagBag.php

```diff
@@ -70,17 +70,26 @@ private static function checkIntBackedEnumType(string $enumType): void
     }
 
     /**
-     * @param class-string<T> $enumType
+     * @template E of \BackedEnum
+     *
+     * @param class-string<E> $enumType
      *
-     * @return FlagBag<T>
+     * @return FlagBag<E>
      */
     public static function fromAll(string $enumType): FlagBag
     {
         return new FlagBag($enumType, static::getBitmask($enumType));
     }
 
     /**
-     * @param class-string<T>|T $enumOrType
+     * @template E of \BackedEnum
+     *
+     * @param class-string<E>|E $enumOrType
+     * @param E                 ...$flags
+     *
+     * @return FlagBag<E>
+     *
+     * @phpstan-return FlagBag<T>
```

Comment:
Why would this one be required here while not in `fromAll` ?

```suggestion
```

Reply:
No idea.

But if this is not present, PHP Stan compain about this:

```
------ ----------------------------------------------------------------------------------------------------------------------------------------------
  Line   FlagBag.php
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------
  93     PHPDoc tag @return with type Elao\Enum\FlagBag<E of BackedEnum> is incompatible with native type static(Elao\Enum\FlagBag<T of BackedEnum>).
 ------ ----------------------------------------------------------------------------------------------------------------------------------------------
```

Reply:
One more time, for me what is working is:

```php
/**
 * @param class-string<T>|T $enumOrType
 *
 * @phpstan-return FlagBag<T>
 *
 * @psalm-template E of \BackedEnum
 * @psalm-param E ...$flags
 * @psalm-return FlagBag<E>
 */
public static function from(string|\BackedEnum $enumOrType, \BackedEnum ...$flags): static
{
	// ...
}
```